api/opencl_runtime_layer.asciidoc

```diff
@@ -14129,7 +14129,10 @@ returned in _errcode_ret_:
     specified by {CL_DEVICE_COMMAND_BUFFER_REQUIRED_QUEUE_PROPERTIES_KHR}.
   * {CL_INVALID_CONTEXT} if all the command-queues in _queues_ do not have
     the same OpenCL context.
-  * {CL_INVALID_VALUE} if _num_queues_ is zero.
+  * {CL_INVALID_VALUE} if the `<<cl_khr_command_buffer_multi_device>>`
+    extension is supported and _num_queues_ is zero, or if the
+    `<<cl_khr_command_buffer_multi_device>>` extension is not supported
+    and _num_queues_ is not one.
```

Comment:
Maybe another way to ask this question is: how much configurability do we expect to have with respect to enabling and disabling different KHR extensions?  We have asciidoctor defines for each individual extension, but should we engineer the spec such that each individual extension can be enabled or disabled individually?  Or, should we expect that all KHR extensions will be enabled as a group, in which case an error code definition like this one would be OK?

Reply:
I don't think there's any harm in the occasional 'if {extension} is enabled ... is disabled ...' like this, without asciidoctor conditionals. All that will happen in the core spec is that the link to the extension name will be dead.

I tried to make the markup I wrote up make sense with or without any particular extension enabled at spec build time. IME with Vulkan, there are not many cases where multiple extensions are fighting over the same bit of language, so it has been tractable. I believe there was one case where I had a little bit of trouble with an error condition like this, and restructured the errors slightly into a 'regular' and a 'if cl_khr_blah is enabled' section, though I cannot recall just where that was so it's hard to point to it as an example even if it's relevant - it may have been more a case of 'this extension adds these errors'.

Reply:
Resolving this as I'm happy with how this currently reads even if the link will be dead without `ifdef::cl_khr_command_buffer_multi_device`.